### PR TITLE
Slot Machine Refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactors the Slot Machine

- Slot machine no longer has an internal cash amount it draws from; it just generates money out of thin air
- Slot machine can no longer be spam played; you'll have to wait until the counter is done spinning before playing again.
- Time from spin to results lowered from 10 seconds to 3 seconds.
- Added in spinning sound for when you start the spinning process
- Added in sound feedback each time you win
- Jackpot raised from 25,000 to the original 250,000
 - Jackpot has its own unique sound compared to the rest
- Absolute pathing
- Spans
- Visible messages instead of show messages

I'm terrible at UI stuff--ideally, that would be updated so that things aren't tied to attack_hand and the proper "please wait" is displayed when you hit "play".

### CODE REVIEW

File: code/game/machinery/slotmachine.dm [hunk 1]
```diff
@@ -6,77 +7,73 @@ var/datum/announcement/minor/slotmachine_announcement = new(do_log = 0)
...
+			usr.mind.initial_account.money -= 5
+			plays += 1
+			working = 1
+			icon_state = "slots-on"
+			var/roll = rand(1,10000)
+			playsound(src.loc, 'sound/machines/ding.ogg', 50, 1)
+			spawn(30)
+				if (roll == 1)
+					playsound(src.loc, 'sound/effects/engine_alert2.ogg', 50, 0)
+					for(var/mob/O in hearers(src, null))
```
Comment (tigercat2000) [inline code/game/machinery/slotmachine.dm]: should be replaced with visible_message

File: code/game/machinery/slotmachine.dm [hunk 2]
```diff
@@ -6,77 +7,73 @@ var/datum/announcement/minor/slotmachine_announcement = new(do_log = 0)
...
+	if(href_list["ops"])
+		var/operation = text2num(href_list["ops"])
+		if(operation == 1) // Play
+			if (working == 1)
+				usr << "<span class='warning'>You need to wait until the machine stops spinning!</span>"
+				return
+			if (balance < 5)
+				usr << "<span class='warning'>Insufficient money to play!</span>"
+				return
+			usr.mind.initial_account.money -= 5
```
Comment (tigercat2000) [inline code/game/machinery/slotmachine.dm]: This seems like the worst possible way to do this (i know it's not your code)

Comment (Fox-McCloud) [inline code/game/machinery/slotmachine.dm]: It originally pulled money off of your card, but then we got VG's economy system or something like that and Ponies changed it to this--I'm not the most familiar with monetary transaction and how ATMs+accounts/etc. works, so I'd ratherrrr not touch this, to be honest.

File: code/game/machinery/slotmachine.dm [hunk 3]
```diff
@@ -6,77 +7,70 @@ var/datum/announcement/minor/slotmachine_announcement = new(do_log = 0)
...
-						usr << "\red No luck!"
-					src.working = 0
-					src.icon_state = "slots-off"
-		src.add_fingerprint(usr)
-		src.updateUsrDialog()
-		return
+/obj/machinery/slot_machine/Topic(href, href_list)
+	if(href_list["ops"])
+		var/operation = text2num(href_list["ops"])
+		if(operation == 1) // Play
```
Comment (Tastyfish) [inline code/game/machinery/slotmachine.dm]: I never understood why it converts the value of `href_list["ops"]` into a number and checks it; isn't `1` the only possible value?

File: code/game/machinery/slotmachine.dm (hunk 2 above)
Comment (Tastyfish) [inline code/game/machinery/slotmachine.dm]: It's pretty bloody complicated, but doing it right adds transaction logs in the person's account, etc, could be good for tracking stolen card.
* First, you get the account properly:
```
var/datum/money_account/A = user.get_worn_id_account()
```
* And then charge it!
```
if(!A)
  ~~ handle no account ~~!
if(!A.charge(5, transaction_purpose="Single Pull",dest_name=name))
  ~~ handle no dosh ~~!
```
 * That assumes you're charging 5 credits and that you want "Single Pull" on the statement for what the person purchased.

Comment (Tastyfish) [inline code/game/machinery/slotmachine.dm]: Oh, and you could pass (1) to get_worm_id_account to require pin entry but I think we all agree that that would be awful for slots.

**Comment by Tastyfish** [comment]
Oh and RIP pulling the slots 2000 times in a round and having it ending up with large negative balances.

File: code/game/machinery/slotmachine.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/game/machinery/slotmachine.dm]: If you'd like to do that in a future PR, be my guest, but I'd rather not do it in this one.

Comment (Tastyfish) [inline code/game/machinery/slotmachine.dm]: :+1:

**Comment by FalseIncarnate** [comment]
I wouldn't mind seeing the slot machine actually get a GUI, might do that later myself (after my current project, of course)

File: code/game/machinery/slotmachine.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/game/machinery/slotmachine.dm]: @Tastyfish Is there any other way I can get in contact with you?

Comment (Tastyfish) [inline code/game/machinery/slotmachine.dm]: @Fox-McCloud Forum PM (tastyfish), byond pager I guess (Tastyfish); you can contact me one of those ways if you prefer steam or skype or something.

**Pull request merged**